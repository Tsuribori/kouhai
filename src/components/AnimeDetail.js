import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import { Card, CardActionArea, CardContent, CardMedia, Paper, Grid, List, ListItem, ListItemIcon, ListItemText, Typography, Link } from '@material-ui/core';
import { StarRate, Movie, LiveTv, Tv, AccessTime, Info, Style, Business, PlayArrow, DateRange } from '@material-ui/icons';
import moment from 'moment';

const styles = theme => ({
  nested: { 
    paddingLeft: theme.spacing.unit * 10,
  },
});

const formatDate = (props) => {
  if (props.aired.from === null) {
    return null;
  }
  else if (props.type === 'Movie') {
    return moment(props.aired.from).format('YYYY/MM/DD');
  }
  else {
    return `${moment(props.aired.from).format('YYYY/MM/DD')} - ${props.aired.to ? moment(props.aired.to).format('YYYY/MM/DD') : 'Ongoing'}`;
  }
};

const info = (props) => [
  {
    'name': 'score',
    'icon': <StarRate/>,
    'value': props.score,
  },
  {
    'name': 'type',
    'icon': <Movie/>,
    'value': props.type,
  },
  {
    'name': 'episodes',
    'icon': <LiveTv/>,
    'value': props.episodes,
  },
  {
    'name': 'status',
    'icon': <Tv/>,
    'value': props.status,
  },
  {
    'name': 'aired',
    'icon': <DateRange/>,
    'value': formatDate(props),
  },
  {
    'name': 'broadcast',
    'icon': <AccessTime/>,
    'value': props.broadcast,
  },
  {
    'name': 'studio',
    'icon': <Business/>,
    'value': props.studios.length > 0 ? props.studios[0].name : null,
  },
];
   
function AnimeDetail(props) {
   return (
     <div> 
     <Card>
       <CardActionArea>
         <Grid
           container
           justify="center"
         >
           <Grid item>
             <CardMedia
               component="img"
               image={props.image_url}
               title={props.title}
             />
           </Grid>
           <Grid item>
             <CardContent>
               <Typography gutterBottom variant="h5" component="h3">
                 {props.title}
               </Typography>
               <Typography variant="subtitle1" color="textSecondary">
                 {props.english_title}
               </Typography>
             </CardContent>
           </Grid>
         </Grid>
       </CardActionArea>
     </Card>
     
     <Paper>      
     <List>
       {info(props).map((obj) => (
         obj.value && (
           <ListItem key={obj.name}>
             <ListItemIcon>
               {obj.icon}
             </ListItemIcon>
             <ListItemText
               primary={obj.value}
               secondary={obj.name}
             />
           </ListItem>
         )
       ))}
       { props.synopsis &&
       <ListItem alignItems="flex-start">
         <ListItemIcon>
           <Info />
         </ListItemIcon>
         <ListItemText
           primary={props.synopsis}
           secondary="synopsis"
         />
       </ListItem>
       }
       <ListItem>
         <ListItemIcon>
           <PlayArrow/>
         </ListItemIcon>
         <ListItemText>
           <Link 
             target="_blank"
             rel="noreferrer"
             href={props.trailer_url}
           >
             View trailer
           </Link>
         </ListItemText>
       </ListItem>
       <ListItem>
         <ListItemIcon>
           <Style />
         </ListItemIcon>
         <ListItemText
           primary="Genres"
         />
       </ListItem>
         <List className={props.classes.nested}>
           {props.genres.map((genre) => (
             <ListItem key={genre.name}>
               <ListItemText primary={genre.name} />
             </ListItem>
           ))}
         </List>
     </List>
     </Paper>
     </div>
   );
}

AnimeDetail.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(AnimeDetail);
