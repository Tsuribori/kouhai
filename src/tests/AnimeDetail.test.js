import React from 'react';
import './TestSetUp.js';
import AnimeDetail from '../components/AnimeDetail.js';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const path = require('path');
const fs = require('fs');

describe('AnimeDetail test', () => {
  test('Test AnimeDetail snapshot', () => {
    const data = JSON.parse(fs.readFileSync(path.resolve(__dirname, './Detail.json')));
    const tree = shallow(<AnimeDetail {...data} />).dive();
    expect(toJson(tree)).toMatchSnapshot();
  });
});
