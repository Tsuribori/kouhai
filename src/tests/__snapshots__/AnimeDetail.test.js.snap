// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimeDetail test Test AnimeDetail snapshot 1`] = `
<div>
  <WithStyles(Card)>
    <WithStyles(CardActionArea)>
      <WithStyles(Grid)
        container={true}
        justify="center"
      >
        <WithStyles(Grid)
          item={true}
        >
          <WithStyles(CardMedia)
            component="img"
            image="https://cdn.myanimelist.net/images/anime/4/19644.jpg"
            title="Cowboy Bebop"
          />
        </WithStyles(Grid)>
        <WithStyles(Grid)
          item={true}
        >
          <WithStyles(CardContent)>
            <WithStyles(Typography)
              component="h3"
              gutterBottom={true}
              variant="h5"
            >
              Cowboy Bebop
            </WithStyles(Typography)>
            <WithStyles(Typography)
              color="textSecondary"
              variant="subtitle1"
            />
          </WithStyles(CardContent)>
        </WithStyles(Grid)>
      </WithStyles(Grid)>
    </WithStyles(CardActionArea)>
  </WithStyles(Card)>
  <WithStyles(Paper)>
    <WithStyles(List)>
      <WithStyles(ListItem)
        key="score"
      >
        <WithStyles(ListItemIcon)>
          <pure(StarRateIcon) />
        </WithStyles(ListItemIcon)>
        <WithStyles(ListItemText)
          primary={8.81}
          secondary="score"
        />
      </WithStyles(ListItem)>
      <WithStyles(ListItem)
        key="type"
      >
        <WithStyles(ListItemIcon)>
          <pure(MovieIcon) />
        </WithStyles(ListItemIcon)>
        <WithStyles(ListItemText)
          primary="TV"
          secondary="type"
        />
      </WithStyles(ListItem)>
      <WithStyles(ListItem)
        key="episodes"
      >
        <WithStyles(ListItemIcon)>
          <pure(LiveTvIcon) />
        </WithStyles(ListItemIcon)>
        <WithStyles(ListItemText)
          primary={26}
          secondary="episodes"
        />
      </WithStyles(ListItem)>
      <WithStyles(ListItem)
        key="status"
      >
        <WithStyles(ListItemIcon)>
          <pure(TvIcon) />
        </WithStyles(ListItemIcon)>
        <WithStyles(ListItemText)
          primary="Finished Airing"
          secondary="status"
        />
      </WithStyles(ListItem)>
      <WithStyles(ListItem)
        key="aired"
      >
        <WithStyles(ListItemIcon)>
          <pure(DateRangeIcon) />
        </WithStyles(ListItemIcon)>
        <WithStyles(ListItemText)
          primary="1998/04/03 - 1999/04/24"
          secondary="aired"
        />
      </WithStyles(ListItem)>
      <WithStyles(ListItem)
        key="broadcast"
      >
        <WithStyles(ListItemIcon)>
          <pure(AccessTimeIcon) />
        </WithStyles(ListItemIcon)>
        <WithStyles(ListItemText)
          primary="Saturdays at 01:00 (JST)"
          secondary="broadcast"
        />
      </WithStyles(ListItem)>
      <WithStyles(ListItem)
        key="studio"
      >
        <WithStyles(ListItemIcon)>
          <pure(BusinessIcon) />
        </WithStyles(ListItemIcon)>
        <WithStyles(ListItemText)
          primary="Sunrise"
          secondary="studio"
        />
      </WithStyles(ListItem)>
      <WithStyles(ListItem)
        alignItems="flex-start"
      >
        <WithStyles(ListItemIcon)>
          <pure(InfoIcon) />
        </WithStyles(ListItemIcon)>
        <WithStyles(ListItemText)
          primary="In the year 2071, humanity has colonized several of the planets and moons of the solar system leaving the now uninhabitable surface of planet Earth behind. The Inter Solar System Police attempts to keep peace in the galaxy, aided in part by outlaw bounty hunters, referred to as \\"Cowboys.\\" The ragtag team aboard the spaceship Bebop are two such individuals. Mellow and carefree Spike Spiegel is balanced by his boisterous, pragmatic partner Jet Black as the pair makes a living chasing bounties and collecting rewards. Thrown off course by the addition of new members that they meet in their travels—Ein, a genetically engineered, highly intelligent Welsh Corgi; femme fatale Faye Valentine, an enigmatic trickster with memory loss; and the strange computer whiz kid Edward Wong—the crew embarks on thrilling adventures that unravel each member's dark and mysterious past little by little. Well-balanced with high density action and light-hearted comedy, Cowboy Bebop is a space Western classic and an homage to the smooth and improvised music it is named after. [Written by MAL Rewrite]"
          secondary="synopsis"
        />
      </WithStyles(ListItem)>
      <WithStyles(ListItem)>
        <WithStyles(ListItemIcon)>
          <pure(PlayArrowIcon) />
        </WithStyles(ListItemIcon)>
        <WithStyles(ListItemText)>
          <WithStyles(Link)
            href="https://www.youtube.com/embed/qig4KOK2R2g?enablejsapi=1&wmode=opaque&autoplay=1"
            rel="noreferrer"
            target="_blank"
          >
            View trailer
          </WithStyles(Link)>
        </WithStyles(ListItemText)>
      </WithStyles(ListItem)>
      <WithStyles(ListItem)>
        <WithStyles(ListItemIcon)>
          <pure(StyleIcon) />
        </WithStyles(ListItemIcon)>
        <WithStyles(ListItemText)
          primary="Genres"
        />
      </WithStyles(ListItem)>
      <WithStyles(List)
        className="AnimeDetail-nested-1"
      >
        <WithStyles(ListItem)
          key="Action"
        >
          <WithStyles(ListItemText)
            primary="Action"
          />
        </WithStyles(ListItem)>
        <WithStyles(ListItem)
          key="Adventure"
        >
          <WithStyles(ListItemText)
            primary="Adventure"
          />
        </WithStyles(ListItem)>
        <WithStyles(ListItem)
          key="Comedy"
        >
          <WithStyles(ListItemText)
            primary="Comedy"
          />
        </WithStyles(ListItem)>
        <WithStyles(ListItem)
          key="Drama"
        >
          <WithStyles(ListItemText)
            primary="Drama"
          />
        </WithStyles(ListItem)>
        <WithStyles(ListItem)
          key="Sci-Fi"
        >
          <WithStyles(ListItemText)
            primary="Sci-Fi"
          />
        </WithStyles(ListItem)>
        <WithStyles(ListItem)
          key="Space"
        >
          <WithStyles(ListItemText)
            primary="Space"
          />
        </WithStyles(ListItem)>
      </WithStyles(List)>
    </WithStyles(List)>
  </WithStyles(Paper)>
</div>
`;
